client/src/hooks/fuse/index.ts

```diff
@@ -14,6 +15,14 @@ const useFuse: <T>(data: T[], options?: Fuse.IFuseOptions<T>) => UseFuseReturn<T
 ) => {
   const [term, setTerm] = useState('');
 
+  const search = useCallback<UseFuseReturn<unknown>['search']>((search) => {
+    if (typeof search === 'string') {
+      setTerm(search);
+    } else {
+      setTerm(search.target.value);
+    }
+  }, []);
```

Comment:
I think this part could be improved, you are assuming things. Let's leave it as it is but add a comment to improve in the future.